js/date-filter.js

```diff
@@ -125,6 +146,127 @@ DateFilter.prototype.handleModifyEvent = function(e, opts) {
   }
 };
 
+DateFilter.prototype.setupDateGrid = function() {
+  var dateBegin = this.$minDate.val().split('/');
+  var dateEnd = this.$maxDate.val().split('/');
+  var dateRangeFirst = this.$grid.find('.date-range-first');
+  var dateRangeSecond = this.$grid.find('.date-range-second');
+  var minDateMonth = dateBegin[0];
+  var minDateYear = dateBegin[2];
+  var maxDateMonth = dateEnd[0];
+  var maxDateYear = dateEnd[2];
+  var transactionYear = $('#two-year-transaction-period').val();
+  var $dateBegin;
+  var $dateEnd;
+
+  // the transaction year starts with the later, so begin before
+  dateRangeFirst.find('.date-range-year').html(parseInt(transactionYear - 1));
+  dateRangeFirst.find('ul').attr('data-year', parseInt(transactionYear - 1));
+
+  dateRangeSecond.find('.date-range-year').html(transactionYear);
+  dateRangeSecond.find('ul').attr('data-year', transactionYear);
+
+  // get the elements of the beginning and ending dates
+  $dateBegin = this.$grid.find('ul[data-year="' + minDateYear + '"] ' +
+    'li[data-month="' + minDateMonth + '"]').addClass('begin');
```

Comment:
Can we make these `begin` and `end` classes more specific? Even just `.month--begin` / `.month--end`